### ISSUE TITLE
"Preferred" fields should only be exempt from updates if Admin makes changes

### ISSUE THREAD

**Issue by karmour** [bcgov/ExitSurveyAdmin#152]
The "Preferred" fields were designed as a way for Admins to overwrite incoming data from the CSVs for certain fields such as name, address, and email. Even though the non-preferred variables might be updated in daily CSV uploads, the Preferred fields would only be editable by the Admin, thus preserving any user-requested changes conveyed to the Admin. 

However, at present the "Preferred" fields are being pre-populated with initial data from the non-preferred fields during the record creation process. This initial data then acts as if it were entered by the Admin and is not updated by CSVs, meaning that even if the Admin makes no changes to the record, the initial data pushed to the Preferred fields are never overwritten by future 
updates from the CSVs.

Ideal behaviour is that the Preferred fields should only be locked from CSV edits if the Admin has made changes. This may be achievable with some kind of flag variable, or with a third related variable that has some logic attached (i.e., use Preferred entry if not blank, else use Non-preferred entry]).

Note: depending on the chosen solution, changes here may require display modifications within the ESA and to data linkages with Callweb to use the updated variable.

**Comment by hangler** [comment]
Will be adding flags for this. Just noting this for future reference, but the command to generate the SQL to move to this migration is:

`dotnet ef migrations script InitialCreate AddPreferredFlagsToEmployee`

**Comment by hangler** [comment]
@karmour @aclowery A flag has now been added to the database that indicates when a preferred field has been overridden by an admin. If the flag is false, the preferred value will be updated to match whatever the "non-preferred" value is. If the flag is true (i.e. the admin has set the value manually at any point), that field will no longer be overridden. (The "non-preferred" value will still be updated, however, as you'd expect.)

I've also added a little icon to the interface to indicate whether the field has been overridden. It's a maybe a little bit overly literal at the moment... let me know if something else would be nicer.

<img width="306" alt="Screen Shot 2020-08-18 at 11 08 56 AM" src="https://user-images.githubusercontent.com/3538990/90549373-41ba2c80-e143-11ea-9488-fa721095759c.png">
